### PR TITLE
Add PKCE parameters to UM auth flow

### PR DESCRIPTION
**Pull request by awolfden**
## Description
Add PKCE parameters to UM auth flow

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[X] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/user_management.py [hunk 1]
```diff
@@ -476,6 +481,9 @@ def get_authorization_url(
             params["login_hint"] = login_hint
         if state is not None:
             params["state"] = state
+        if code_challenge is not None:
```
Comment (ameesha) [inline workos/user_management.py:484]: is an empty string valid? if not you can switch this to `if code_challenge:`

File: workos/user_management.py [hunk 2]
```diff
@@ -444,6 +446,9 @@ def get_authorization_url(
                 OktaSAML, and AzureSAML connection types. (Optional)
             state (str) - An encoded string passed to WorkOS that'd be preserved through the authentication workflow, passed
                 back as a query parameter. (Optional)
+            code_challenge (str) - Code challenge is derived from the code verifier used for the PKCE flow. (Optional)
+            code_challenge_method ("S256") - The only valid PKCE code challenge method is "S256".
```
Comment (ameesha) [inline workos/user_management.py:450]: if there's only 1 option for this field, why pass it in at all? let's just set it only when code_challenge is passed in

File: workos/user_management.py (hunk 1 above)
Comment (awolfden) [reply workos/user_management.py]: Updated ✅

File: workos/user_management.py (hunk 2 above)
Comment (awolfden) [reply workos/user_management.py]: Updated ✅

File: workos/user_management.py [hunk 3]
```diff
@@ -476,6 +478,9 @@ def get_authorization_url(
             params["login_hint"] = login_hint
         if state is not None:
             params["state"] = state
+        if code_challenge:
+            params["code_challenge"] = code_challenge
+            params["code_challenge_method"] = "S256"
```
Comment (ameesha) [inline workos/user_management.py:483]: quick follow up - since this is a static value, does the client need to pass it in at all?

**Review by ameesha — APPROVED** [review]

File: workos/user_management.py (hunk 3 above)
Comment (awolfden) [reply workos/user_management.py:483]: It's listed in the API reference as a param for curl, so I included it here

**Pull request merged**